### PR TITLE
Implement plugin dependencies, loaded in dependency first order

### PR DESCRIPTION
**Pull request by NotStirred**
This PR doesn't implement a plugin loader API but that is now ready for another PR.

Outstanding things:
- [x] Added one dependency `org.apache.maven:maven-artifact:3.9.6` (Apache 2)
- [x] #1702
- [x] Tests for plugin load order

### CODE REVIEW

File: chunky/src/java/se/llbit/chunky/main/Chunky.java [hunk 1]
```diff
@@ -265,35 +263,31 @@ private void loadPlugins() {
...
+    // TODO: allow plugins to implement a custom plugin loader.
+    PluginManager pluginManager = new PluginManager(new JarPluginLoader());
+
+    // Parse plugin manifests
+    Set<PluginManifest> pluginManifests = plugins.elements.stream()
+      .map(value -> value.asString(""))
+      .filter(jarName -> !jarName.isEmpty())
+      .map(jarName -> pluginsPath.resolve(jarName).toAbsolutePath().toFile())
+      .map(PluginManager::parsePluginManifest)
+      .flatMap(o -> o.map(Stream::of).orElseGet(Stream::empty)) // replace with Optional::stream in java9+
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/main/Chunky.java:275]: We're on Java 17 now

File: chunky/src/java/se/llbit/chunky/plugin/loader/JarPluginLoader.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+public class JarPluginLoader implements PluginLoader {
+  public void load(BiConsumer<Plugin, PluginManifest> onLoad, PluginManifest pluginManifest) {
+    try {
+      Class<?> pluginClass = loadPluginClass(pluginManifest.main, pluginManifest.pluginJar);
+      Plugin plugin = (Plugin) pluginClass.getDeclaredConstructor().newInstance();
+      onLoad.accept(plugin, pluginManifest);
+    } catch (IOException | ClassNotFoundException e) {
+      Log.error("Could not load the plugin", e);
+    } catch (ClassCastException e) {
+      Log.error("Plugin main class has wrong type", e);
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/plugin/loader/JarPluginLoader.java:26]:
```suggestion
      Log.error("Plugin main class has wrong type (must implement se.llbit.chunky.Plugin)", e);
```

File: chunky/src/java/se/llbit/chunky/plugin/loader/JarPluginLoader.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    } catch (InstantiationException | IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
+      Log.error("Could not create plugin instance", e);
+    }
+  }
+
+  /**
+   * This method is {@link PluginApi} to allow plugins to override only classloading functionality of the default plugin loader.
```
Comment (leMaik) [inline chunky/src/java/se/llbit/chunky/plugin/loader/JarPluginLoader.java:33]: Is there an actual use-case for overriding the plugin loader?

**Review by leMaik — CHANGES_REQUESTED** [review]
Looks good, just unsure whether we need the plugin loader to be pluggable.

**Comment by NotStirred** [comment]
Yes that is one of the main goals for this PR, do you remember the discussion about having a plugin that would transform 2.4.x plugins to support 2.5.x etc.? That's the main reason

**Comment by leMaik** [comment]
Ah, indeed I do remember this. Allright. :+1:

**Review by leMaik — APPROVED** [review]

**Comment by leMaik** [comment]
I added the new dependency to the readme etc. and replaced semver4j with maven-artifact. What do you think about the last two commits, @NotStirred?

**Comment by NotStirred** [comment]
Chunky is not and never has been semver, and plugins are almost definitely not either

**Comment by NotStirred** [comment]
new commits look good, thanks 👍

**Pull request merged**